Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Promact.Trappist.DomainModel.Models.Category;
+using Promact.Trappist.Repository.Categories;
+
+namespace Promact.Trappist.Core.Controllers
+{
+    [Route("api/category")]
+    public class CategoryController : Controller
+    {
+        private readonly ICategoryRepository _categoryRepository;
+        public CategoryController(ICategoryRepository categoryRepository)
+        {
+            _categoryRepository = categoryRepository;
+        }
+        /// <summary>
+        /// Get All Categories
+        /// </summary>
+        /// <returns>Category List</returns>
+        [HttpGet]
+        public IActionResult GetAllCategories()
+        {
+            var categoryList = _categoryRepository.GetAllCategories();
+            return Ok(categoryList);
+        }
+
+        #region post Method 
+        [HttpPost]
+        /// <summary>
+        /// Post Method 
+        /// Will Add a Catagory Name in Category Model
+        ///</summary>
+        /// <param name="category">Object of  class Category</param>
+        /// <returns>object of the class </returns>
+        public IActionResult catagoryAdd([FromBody] Category category)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename method 'catagoryAdd' to match camel case naming rules, consider using 'CatagoryAdd'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS100)